app/workers/deferred_retraction.rb

```diff
@@ -0,0 +1,18 @@
+#   Copyright (c) 2010-2011, Diaspora Inc.  This file is
+#   licensed under the Affero General Public License version 3 or later.  See
+#   the COPYRIGHT file.
+
+module Workers
+  class DeferredRetraction < Base
+    sidekiq_options queue: :dispatch
+
+    def perform(user_id, retraction_data, recipient_ids, opts)
+      user = User.find(user_id)
+      subscribers = Person.where(id: recipient_ids)
+      object = Retraction.new(retraction_data.deep_symbolize_keys, subscribers)
+      opts = HashWithIndifferentAccess.new(opts)
```

Comment:
Okay this probably a bit too fundamental at this point, but I would have loved for the lower level to not rely on HWIAs and expect normalized to string keys option hashes. Or given Ruby 2.3's Symbol GC we could even consider normalizing to symbols.